crates/block-producer/src/mempool/tests.rs

```diff
@@ -16,6 +16,29 @@ impl Mempool {
     }
 }
 
+#[tokio::test]
+async fn mempool_trace() {
+    let (mut rx_export, _rx_shutdown) = miden_node_utils::logging::setup_test_tracing().unwrap();
+
+    let mut uut = Mempool::for_tests();
+    let txs = MockProvenTxBuilder::sequential();
+    uut.add_transaction(txs[0].clone()).unwrap();
+
+    let span_data = rx_export.recv().await.unwrap();
+    assert_eq!(span_data.name, "mempool.add_transaction");
+    assert_eq!(
+        span_data
+            .attributes
+            .iter()
+            .filter(|kv| {
+                kv.key == "code.namespace".into()
+                    && kv.value == "miden_node_block_producer::mempool".into()
+            })
+            .count(),
+        1,
+    );
+}
```

Comment:
@Mirko-von-Leipzig I added this unit test to illustrate the otel trace test capability. It doesn't relate to the functionality added in this PR. This was just the easiest unit test I could find to impl. The block producer rpc stack doesn't have mocks etc atm.

LMK if you want to keep/rm this test or move it to another PR with other tests etc.

Reply:
Let's leave it for a separate PR.

Is it possible to separate by test itself - or do we need to run these kind of tests sequentially to ensure we get the spans recorded that we expect?

As in, running tests in parallel probably muddles the span exporter unless we somehow mark the spans with a test ID?

But this looks promising, we can figure out the specifics in the other PR/issue.